### PR TITLE
Removing some raw usages of generic types introduced by our previous selftype refactoring.

### PR DESCRIPTION
**Pull request by VoglSebastian**

### CODE REVIEW

File: master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java [hunk 1]
```diff
@@ -53,37 +53,34 @@
 
 	    EvoSuite evosuite = new EvoSuite();
 
-	    String targetClass = cut;
-	    Properties.TARGET_CLASS = targetClass;
+		Properties.TARGET_CLASS = cut;
```
Comment (schweikl) [inline master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java:56]: Just a cosmetic thing, but still: can you fix the indentation?

File: master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java [hunk 2]
```diff
@@ -53,37 +53,34 @@
...
 	    Object result = evosuite.parseCommandLine(command);
 	    Assert.assertNotNull(result);
 
 	    GeneticAlgorithm<?> ga = getGAFromResult(result);
-	    
-	    List<Chromosome> population = new ArrayList<>(ga.getBestIndividuals());
-	    
-	    return population;
+
+		return new ArrayList<>(ga.getBestIndividuals());
```
Comment (schweikl) [inline master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java:65]: Just a cosmetic thing, but still: can you fix the indentation?

**Review by schweikl — COMMENTED** [review]
Looks good to me so far :+1:

**Comment by schweikl** [comment]
BTW, out of curiosity, you probably noticed there's this method `getGAFromResult(...)` which returns a raw `GeneticAlgorithm` at the moment. Any chances we could refactor this method as well?

**Comment by VoglSebastian** [comment]
> BTW, out of curiosity, you probably noticed there's this method `getGAFromResult(...)` which returns a raw `GeneticAlgorithm` at the moment. Any chances we could refactor this method as well?

getGAFromResult returns a Wildcard type as parameter of GA atm in SystemTestBase, but you probably meant casting it inside that method and returning a GeneticAlgorithm<TestSuiteChromosome>?

I was hesitant to do this, because i wasn't sure if there are any tests that may rely on a GeneticAlgorithm<TestChromosome>?

**Comment by VoglSebastian** [comment]
I have made some changes regarding the Method getGAFromResults.
f42ccd1 is the commit that makes the changes, the later one only updates the call sites.

**Comment by schweikl** [comment]
> getGAFromResult returns a Wildcard type as parameter of GA atm in SystemTestBase, but you probably meant casting it inside that method and returning a GeneticAlgorithm?
> 
> I was hesitant to do this, because i wasn't sure if there are any tests that may rely on a GeneticAlgorithm?

Oh, yes, you are right of course, it currently returns a `GeneticAlgorithm<?>` but it would be nice to have a `GeneticAlgorithm<T>` instead. Sorry for the confusion.

**Review by schweikl — COMMENTED** [review]
With all of those wildcards and downcasts gone, I think the tests look so much better now :+1: Thank you!

File: master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java (hunk 2 above)
Comment (schweikl) [reply master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java]: OK, I saw you fixed the indentation in your latest changes. Thx!

File: master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java (hunk 1 above)
Comment (schweikl) [reply master/src/test/java/org/evosuite/ga/metaheuristics/CellularGASystemTest.java]: OK, I saw you fixed the indentation in your latest changes. Thx!

**Comment by schweikl** [comment]
Some of the tests on CircleCI fail, but AFAIK they have been failing way before your MR. So I would be fine with that and happy to merge your changes.

**Comment by gofraser** [comment]
Looks great, thanks!

**Pull request merged**